### PR TITLE
Throw in `chain()` if account is not passed nor previously recorded

### PR DESCRIPTION
**Pull request by gtupak**
Improves data quality.

### CODE REVIEW

File: src/ArcxAnalyticsSdk.ts [hunk 1]
```diff
@@ -612,6 +612,12 @@ export class ArcxAnalyticsSdk {
       throw new Error('ArcxAnalyticsSdk::chainChanged: chainId cannot be empty or 0')
     }
 
+    if (!account && !this.currentConnectedAccount) {
+      throw new Error(
+        'ArcxAnalyticsSdk::chainChanged: account was empty and no previous account has been recorded. You can either pass an account or call wallet() first',
+      )
+    }
+
```
Comment (classicus-eth) [inline src/ArcxAnalyticsSdk.ts:620]: Is throwing an error to aggressive? Should we log an error instead?

Just thinking out loud, not sure as I lack context. 

Also, docstring of this function should be updated if we maintain throwing the error 👍🏻

**Review by classicus-eth — DISMISSED** [review]
Comment to double check something, nothing blocking🥇

File: src/ArcxAnalyticsSdk.ts (hunk 1 above)
Comment (gtupak) [reply src/ArcxAnalyticsSdk.ts:620]: This new version of the SDK throws errors in the public functions to ensure the clients pass the required data to them. I updated the docstring (plus found some other incorrect error labels that I fixed)

**Review by classicus-eth — APPROVED** [review]

**Pull request merged**